lib/tapyrus/rpc/tapyrus_core_client.rb

```diff
@@ -4,6 +4,33 @@
 module Tapyrus
   module RPC
 
+    # Throw when happened anything http's error with connect to server.
+    #
+    # Almost case this exception happened from 401 Unauthorized.
+    # And also, throw by cause of other http's errors.
+    #
+    # You can get raw response as cause, like as:
+    #
+    # rescue TapyrusClientConnectionError => ex
+    #   ex.response
+    # end
+    class ConnectionError < StandardError
```

Comment:
```suggestion
    class Error < StandardError
```

In this case, there are some possible errors other than 401, such as 500, and the connection is well, so the name `ConnectionError` is inappropriate.

Reply:
Agree. you are correct.